ci/github/coverity.sh

```diff
@@ -0,0 +1,53 @@
+#! /bin/bash
+#
+# Copyright 2017 - 2022 James E. King III
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE_1_0.txt or copy at
+#      http://www.boost.org/LICENSE_1_0.txt)
+#
+# Bash script to run in GHA to perform a Coverity Scan build
+# To skip the coverity integration download (which is huge) if
+# you already have it from a previous run, add --skipdownload
+#
+
+#
+# Environment Variables (you must define)
+#
+# COVERITY_SCAN_NOTIFICATION_EMAIL  - email address to notify
+# COVERITY_SCAN_TOKEN               - the Coverity Scan token (should be secure)
+#
+
+set -ex
+
+. "$(dirname "${BASH_SOURCE[0]}")"/../enforce.sh
+: ${B2_TARGETS:="libs/$SELF/test"}
+: ${CLANG_COMPILER:="$(echo ${B2_COMPILER} | cut -d'-' -f1)++-$(echo ${B2_COMPILER} | cut -d'-' -f2)"}
```

Comment:
We have run `install.sh` already. So this is simply `$CXX`, isn't it?
See https://github.com/boostorg/boost-ci/blob/39c21a03b7488ce2e2114cf8888e8135bdfe06c6/ci/common_install.sh#L116

Reply:
Yes, but then the comptype needs to be set.  I'll look at supporting gcc, but I wasn't planning on needing to do it.

Reply:
Sure, but all you do here is a (complicated) way of replacing "clang-xxx" by "clang++-xxx", don't you? This is already done and present in $CXX, so you can use that. As for comptype I suggested a check&error that CXX is clang, which is would be required with this solution too. I mean you could have B2_COMPILER=gcc-10 here, couldn't you?

Reply:
CXX is only valid during the install.sh phase; it is not written back to GITHUB_ENV therefore is is not present later on (nor do I think we want to set it globally in the environment, let b2 do what is normally does here....  We write clang++-10 in this case to the user-profiles.jam file though.